.github/workflows/canary-integration-test.yml

```diff
@@ -1252,6 +1252,49 @@ jobs:
+        run: |
+          kubectl exec -n rook-ceph-secondary deploy/rook-ceph-tools -t -- ceph fs subvolume create myfs testsubvolume
+
+      - name: Deploy Direct Tools pod on primary cluster
+        run: |
+          kubectl -n rook-ceph create -f deploy/examples/direct-mount.yaml
+
+      - name: Deploy Direct Tools pod on secondary cluster
+        run: |
+          kubectl -n rook-ceph-secondary create -f deploy/examples/direct-mount.yaml
+
+      - name: Configure a directory for snapshot mirroring on primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror enable myfs
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs snapshot mirror add myfs /volumes/_nogroup/testsubvolume/
+
+      - name: make sure that snapshot mirror is enabled on the secondary cluster
+        run: |
+          kubectl -n rook-ceph-secondary deploy/rook-ceph-tools -t -- ceph fs snapshot mirror enable myfs
+
+      - name: Create 3 snapshots on cluster primary cluster
+        run: |
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap1
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap2
+          kubectl exec -n rook-ceph deploy/rook-ceph-tools -t -- ceph fs subvolume snapshot create myfs testsubvolume snap3
+
+      - name: Get the peer and verify the peer synchronization status that snaps have synced on secondary cluster
+        run: |
+          mirror_daemon=$(kubectl -n rook-ceph exec -it deploy/rook-ceph-fs-mirror -- ls /var/run/ceph/|grep "fs-mirror"| head -n 1)
+          clusterfsid=$(kubectl -n rook-ceph exec -it deploy/rook-ceph-fs-mirror -- ceph --admin-daemon /var/run/ceph/$mirror_daemon fs mirror status myfs@1 |jq '.peers|keys[]')
+          snaps=$(kubectl -n rook-ceph exec -it deploy/rook-ceph-fs-mirror -- ceph --admin-daemon /var/run/ceph/$mirror_daemon fs mirror peer status myfs@1 $clusterfsid|jq '."/volumes/_nogroup/testsubvolume"."snaps_synced"')
+          kubectl -n rook-ceph-secondary exec deploy/rook-direct-mount -t -- mkdir /tmp/registry
+          mon_endpoints=$(kubectl -n rook-ceph-secondary exec deploy/rook-direct-mount -t -- grep mon_host /etc/ceph/ceph.conf | awk '{print $3}')
+          my_secret=$(kubectl -n rook-ceph-secondary exec deploy/rook-direct-mount -t -- grep key /etc/ceph/keyring | awk '{print $3}')
+          kubectl -n rook-ceph-secondary exec deploy/rook-direct-mount -t -- mount -t ceph -o mds_namespace=myfs,name=admin,secret=$my_secret $mon_endpoints:/ /tmp/registry
+          num_snaps_target=$(kubectl -n rook-ceph-secondary exec deploy/rook-direct-mount -t -- ls -lhsa /tmp/registry/volumes/_nogroup/testsubvolume/.snap|grep snap|wc -l)
+          if [ $num_snaps_target = $snaps ]; then echo "Snaphots have synced."; else echo "Snaps have not synced"; fi
```

Comment:
since this the snap sync verification checkpoint, should we be making it "Snap have not synced" fatal?

Reply:
sure.